revision/internal/revision/scanner.go

```diff
@@ -0,0 +1,112 @@
+package revision
```

Comment:
the package should be in the root of the package inside of a internal folder so:
`revision/internal/revision/` ->  `internal/revision/`